src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/ZipVerifiers.java

```diff
@@ -125,32 +113,41 @@ static ZipInputStream sha256WithRsaVerification(ZipStreamWithSignature zipWithSi
         }
     }
 
-
-    static boolean verifyFileNames(Map<String, byte[]> entries) {
-        return
+    static void verifyFileNames(Map<String, byte[]> entries) {
+        boolean verifiedPositively =
             entries.size() == 2
             &&
             entries.keySet().stream()
             .filter(n -> DOCUMENTS_ZIP.equalsIgnoreCase(n) || SIGNATURE_SIG.equalsIgnoreCase(n))
             .count() == 2;
+
+        if (!verifiedPositively) {
+            throw new DocSignatureFailureException(
+                "Zip entries do not match expected file names. Actual names = " + entries.keySet()
+            );
+        }
     }
 
-    private static boolean verifySignature(String publicKeyBase64, Map<String, byte[]> entries) {
-        return verifySignature(
+    private static void verifySignature(String publicKeyBase64, Map<String, byte[]> entries) {
+        verifySignature(
             publicKeyBase64,
             entries.get(DOCUMENTS_ZIP),
             entries.get(SIGNATURE_SIG)
         );
     }
 
-    public static boolean verifySignature(String publicKeyBase64, byte[] data, byte[] signed) {
+    public static void verifySignature(String publicKeyBase64, byte[] data, byte[] signed) {
         PublicKey publicKey = decodePublicKey(publicKeyBase64);
         try {
             Signature signature = Signature.getInstance("SHA256withRSA");
             signature.initVerify(publicKey);
             signature.update(data);
-            return signature.verify(signed);
-        } catch (NoSuchAlgorithmException | InvalidKeyException | SignatureException e) {
+            if (!signature.verify(signed)) {
+                throw new DocSignatureFailureException(INVALID_SIGNATURE_MESSAGE);
+            }
+        } catch (SignatureException e) {
+            throw new DocSignatureFailureException(INVALID_SIGNATURE_MESSAGE, e);
```

Comment:
This case is the reason why I had to make this method throw `DocSignatureFailureException` (with cause attached), instead of returning success/failure.